### PR TITLE
Update CAPI models and client

### PR DESCRIPTION
**Pull request by wpf500**
CAPI models have undergone a major version upgrade, which I think changes a lot of internals, but nothing externally (@tomrf1 might know more, see https://github.com/guardian/content-api-models/pull/60)

I need a more recent version of CAPI models (`10.2`) for the interactive atom work, this PR is a stepping stone to get frontend synced to `10.x`. I'm not aware of any testing, but have spoken to @jfsoul who says he will check it!

### CODE REVIEW

**Comment by dominickendrick** [comment]
I think you accidentally deleted the PR guidelines before you wrote this description 😉 
Can you add a more detailed descriptions about why we need this change ?
Also is there any testing we need to do on this ?

File: project/Dependencies.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ object Dependencies {
   val slf4jVersion = "1.7.5"
   val awsVersion = "1.11.7"
   val faciaVersion = "2.0.2"
-  val capiModelsVersion = "9.24"
+  val capiModelsVersion = "10.1"
```
Comment (dominickendrick) [inline project/Dependencies.scala]: Why is this `10.1` and the client `10.2` ?

Comment (wpf500) [inline project/Dependencies.scala]: The Guardian seems to like messing with people.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ object Dependencies {
   val commonsLang = "commons-lang" % "commons-lang" % "2.4"
   val commonsIo = "commons-io" % "commons-io" % "2.4"
   val cssParser = "net.sourceforge.cssparser" % "cssparser" % "0.9.18"
-  val contentApiClient = "com.gu" %% "content-api-client" % "9.5"
```
Comment (gtrufitt) [inline project/Dependencies.scala]: If this is for the interactive atom, [isn't that in 10.3](https://github.com/guardian/content-api-scala-client/blob/96b8ad38ed3921e69992ee71eb0f14798ab973c1/CHANGELOG.md)?

**Comment by wpf500** [comment]
@dominickendrick, added a better description

**Comment by wpf500** [comment]
It seems the tests are failing but I don't have TeamCity access, @jfsoul happy to help if I can?

File: project/Dependencies.scala (hunk 1 above)
Comment (wpf500) [inline project/Dependencies.scala]: The fact that these version numbers is so similar is in fact coincidental I think, they aren't always necessarily released together. @mchv probably knows more

File: project/Dependencies.scala (hunk 2 above)
Comment (gtrufitt) [inline project/Dependencies.scala]: I commented on the wrong side

Comment (wpf500) [inline project/Dependencies.scala]: I'm trying to separate my PRs, so bumping you up to the `10.x` series first, then will bump to the version with interactive atoms in my atom PR

File: project/Dependencies.scala (hunk 1 above)
Comment (dominickendrick) [inline project/Dependencies.scala]: Ah ok !

**Comment by jfsoul** [comment]
@wpf500 Teamcity has 185 test failures - a lot of these can be put down to test fixtures needing to be updated/added, but there are also a lot that aren't related to this.

I'm not sure how much work there is in fixing all these yet, but it might not be unsubstantial.  Will take a further look tomorrow.

File: project/Dependencies.scala (hunk 1 above)
Comment (dominickendrick) [inline project/Dependencies.scala]: @mchv says the `content-api-models-scala` and `content-api-models-json` are included as a transitive dependency of the `content-api-client`, so I don't think we need to import these specifically, so we should be able to remove these

Comment (wpf500) [inline project/Dependencies.scala]: I tried this but it failed at runtime, so there does seem to be some strange dependency somewhere

Comment (gtrufitt) [inline project/Dependencies.scala]: It's worth pinging @kelvin-chappell (just did) as he recently added the `capiModelsVersion` I believe

**Comment by kelvin-chappell** [comment]
The capi client no longer supports json from v10x, https://github.com/guardian/content-api-scala-client/pull/170, 
so any tests that depend on cached test capi fixtures will fail until the HttpRecorder is modified.

**Comment by kelvin-chappell** [comment]
I need to use the latest models too so I'll get this sorted out.

**Comment by kelvin-chappell** [comment]
Please see #14615

**Comment by wpf500** [comment]
Closing this as fixed by #14615

**Pull request closed without merging**